src/components/Drawer.js

```diff
@@ -0,0 +1,146 @@
+const React = require('react');
+const Velocity = require('velocity-animate');
+
+const StyleConstants = require('../constants/Style');
+
+const Icon = require('../components/Icon');
+
+const Drawer = React.createClass({
+  propTypes: {
+    duration: React.PropTypes.number,
+    easing: React.PropTypes.array,
+    isOpen: React.PropTypes.bool,
+    onClose: React.PropTypes.func.isRequired,
+    siblingContent: React.PropTypes.shape({
+      currentSiblingIndex: React.PropTypes.number.isRequired,
+      nextSibling: React.PropTypes.func.isRequired,
+      previousSibling: React.PropTypes.func.isRequired,
+      totalSiblings: React.PropTypes.number.isRequired
+    })
+  },
+
+  getDefaultProps () {
+    return {
+      duration: 500,
+      isOpen: false,
+      easing: [0.28, 0.14, 0.34, 1.04]
+    };
+  },
+
+  componentDidMount () {
+    this._renderTransition(this.props.isOpen);
+  },
+
+  componentWillReceiveProps (newProps) {
+    if (newProps.isOpen !== this.props.isOpen) {
+      this._renderTransition(newProps.isOpen);
+    }
+  },
+
+  _renderSiblingContent () {
+    const styles = this.styles();
+
+    return (
+      <div ref={(ref) => (this._siblingContent = ref)} style={styles.siblingContent}>
+        <Icon onClick={this.props.siblingContent.previousSibling} size={25} style={styles.icon} type='caret-left'/>
+        {this.props.siblingContent.currentSiblingIndex} of {this.props.siblingContent.totalSiblings}
+        <Icon onClick={this.props.siblingContent.nextSibling} size={25} style={styles.icon} type='caret-right'/>
+      </div>
+    );
+  },
+
+  _renderTransition (isOpen) {
+    const el = this._component;
+    const transition = isOpen ? { right: -800 } : { right: 0 };
+    const options = {
+      complete: this._slideArrowAndSiblingContent.bind(this, isOpen),
+      duration: this.props.duration,
+      easing: this.props.easing
+    };
+
+    Velocity(el, transition, options);
+  },
+
+  _slideArrowAndSiblingContent (isOpen) {
+    this._slideArrow(isOpen);
+    this._slideSiblingContent(isOpen);
+  },
+
+  _slideArrow (isOpen) {
+    const el = this._arrow;
+    const transition = isOpen ? { left: -25 } : { left: 25 };
+    const options = {
+      duration: 200,
```

Comment:
Is there a reason why we use a hard coded value for duration here but use `this.props.duration` elsewhere?